cpp/src/arrow/sparse_tensor.h

```diff
@@ -234,14 +243,16 @@ class ARROW_EXPORT SparseTensorImpl : public SparseTensor {
   // Constructor for empty sparse tensor
   SparseTensorImpl(const std::shared_ptr<DataType>& type,
                    const std::vector<int64_t>& shape,
-                   const std::vector<std::string>& dim_names = {});
-
-  // Constructor with a dense numeric tensor
-  template <typename TYPE>
-  explicit SparseTensorImpl(const NumericTensor<TYPE>& tensor);
+                   const std::vector<std::string>& dim_names = {})
+      : SparseTensorImpl(NULLPTR, type, NULLPTR, shape, dim_names) {}
 
   // Constructor with a dense tensor
-  explicit SparseTensorImpl(const Tensor& tensor);
+  explicit SparseTensorImpl(const Tensor& tensor)
+      : SparseTensorImpl(NULLPTR, tensor.type(), NULLPTR, tensor.shape(),
+                         tensor.dim_names_) {
+    (void)internal::MakeSparseTensorFromTensor(tensor, SparseIndexType::format_id,
```

Comment:
I don't like that we're ignoring the return value here. Perhaps instead `MakeSparseTensorFromTensor` should return ` void` and use `ARROW_CHECK_OK` internally?

Reply:
done